test_getaddrinfo.c

```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+
+#include <assert.h>
+
+#define ASSERT_EQUAL(str1, str2) do { \
+ assert(str1 != NULL); \
+ assert(str2 != NULL); \
+ assert(strcmp(str1, str2)==0); } while (0)
+#define ASSERT assert
+
+void resolve(const char *hostname, char *out) {
+	struct addrinfo hints;
+	struct addrinfo *result;
+
+	memset(&hints, 0, sizeof(hints));
```

Comment:
I don't think this is needed just initialize hints in declaration (e.g. hints = {}).

Reply:
Used `{}` to initialize the fields as well, much cleaner indeed.